File: android/app/src/main/java/com/tiagohs/cinema_history/models/main_topics/MilMoviesMainTopic.kt
package com.tiagohs.cinema_history.models.main_topics

import com.google.gson.annotations.SerializedName
import com.tiagohs.cinema_history.models.image.Image
import java.io.Serializable

class MilMoviesMainTopic(
    @SerializedName("id")
    val id: Int,

    @SerializedName("list_id")
    val list_id: String,

    @SerializedName("title")
    val title: String,

    @SerializedName("background_color")
    val backgroundColor: String? = null,

    @SerializedName("title_color")
    val titleColor: String? = null,

    @SerializedName("image")
    val image: Image
): MainTopic() , Serializable

File: ios/HistoryCinema/HistoryCinema/Entities/LocalFiles/MainTopics/MilMoviesMainTopic.swift
//
//  MilMoviesMainTopic.swift
//  HistoryCinema
//
//  Created by Tiago Henrique da Silva on 27/02/21.
//

import Foundation
import ObjectMapper

class MilMoviesMainTopic: MainTopic {
    var id: Int!
    var listId: String!
    var title: String!
    var backgroundColor: String?
    var titleColor: String?
    var image: LocalImage!
    
    required init?(map: Map) {
        super.init(map: map)
    }

    override func mapping(map: Map) {
        super.mapping(map: map)
        
        id                                              <- map["id"]
        listId                                          <- map["list_id"]
        title                                           <- map["title"]
        backgroundColor                                 <- map["background_color"]
        image                                           <- map["image"]
        titleColor                                      <- map["title_color"]
    }
}
